src/Microsoft.IdentityModel.Clients.ActiveDirectory/Platforms/iOS/WebUI.cs

```diff
@@ -44,6 +49,30 @@ public WebUI(IPlatformParameters parameters)
             {
                 throw new ArgumentException("parameters should be of type PlatformParameters", "parameters");
             }
+
+            DidEnterBackgroundNotification = NSNotificationCenter.DefaultCenter.AddObserver(UIApplication.DidEnterBackgroundNotification, OnMoveToBackground);
+            WillEnterForegroundNotification = NSNotificationCenter.DefaultCenter.AddObserver(UIApplication.WillEnterForegroundNotification, OnMoveToForeground);
+        }
+
+        void OnMoveToBackground(NSNotification notification)
+        {
+            taskId = UIApplication.SharedApplication.BeginBackgroundTask(() => {
+                inBackground = false;
+                UIApplication.SharedApplication.EndBackgroundTask(taskId);
+            });
+
+            //After iOS 11.3, it is neccesary to keep a background task running while moving an app to the background in order to prevent the system from reclaiming network resources from the app. 
+            //This will prevent authentication from failing while the application is moved to the background while waiting for MFA to finish.
+            new Task(() => {
+                inBackground = true;
+                while (inBackground) { }
+                UIApplication.SharedApplication.EndBackgroundTask(taskId);
+            }).Start();
+        }
+
+        void OnMoveToForeground(NSNotification notification)
+        {
+            inBackground = false;
```

Comment:
In iOS we are clearing foreground notification here and we're only listening to foreground transition after app goes to background. We still need to listen for background transition at this point in case user backgrounds the app twice and clean it up when we're done with auth or the class is dealloced.